src/Parallel/ArrayCollection/CreateElementCollection.hpp

```diff
@@ -0,0 +1,201 @@
+      Parallel::Tags::ElementLocations<Dim>, Tags::NumberOfElementsTerminated>;
+  using compute_tags = tmpl::list<>;
+  using const_global_cache_tags =
+      tmpl::list<::domain::Tags::Domain<Dim>,
+                 ::domain::Tags::ElementDistribution>;
+
+  using return_tag_list = tmpl::append<simple_tags, compute_tags>;
+
+  template <typename DbTagsList, typename... InboxTags, typename ArrayIndex,
+            typename ActionList, typename ParallelComponent>
+  static Parallel::iterable_action_return_t apply(
+      db::DataBox<DbTagsList>& box,
+      const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+      Parallel::GlobalCache<Metavariables>& local_cache,
+      const ArrayIndex& /*array_index*/, const ActionList /*meta*/,
+      const ParallelComponent* const /*meta*/) {
+    const std::unordered_set<size_t> procs_to_ignore{};
+
+    const auto& domain = Parallel::get<domain::Tags::Domain<Dim>>(local_cache);
+    const auto& initial_refinement_levels =
+        get<domain::Tags::InitialRefinementLevels<Dim>>(box);
+    const auto& initial_extents = get<domain::Tags::InitialExtents<Dim>>(box);
+    const auto& quadrature = get<evolution::dg::Tags::Quadrature>(box);
+    const std::optional<domain::ElementWeight>& element_weight =
+        Parallel::get<domain::Tags::ElementDistribution>(local_cache);
+
+    const size_t number_of_procs =
+        Parallel::number_of_procs<size_t>(local_cache);
+    const size_t number_of_nodes =
+        Parallel::number_of_nodes<size_t>(local_cache);
+    const size_t num_of_procs_to_use = number_of_procs - procs_to_ignore.size();
+
+    const auto& blocks = domain.blocks();
+
+    const size_t total_num_elements = [&blocks, &initial_refinement_levels]() {
+      size_t result = 0;
+      for (const auto& block : blocks) {
+        const auto& initial_ref_levs = initial_refinement_levels[block.id()];
+        for (const size_t ref_lev : initial_ref_levs) {
+          result += two_to_the(ref_lev);
+        }
+      }
+      return result;
+    }();
+    std::vector<std::pair<ElementId<Dim>, size_t>> my_elements_and_cores{};
+    my_elements_and_cores.reserve(
+        static_cast<size_t>(std::round(static_cast<double>(total_num_elements) /
+                                       static_cast<double>(number_of_nodes))));
+    std::unordered_map<ElementId<Dim>, size_t> node_of_elements{};
+    const size_t my_node = Parallel::my_node<size_t>(local_cache);
+
+    Parallel::create_elements_using_distribution(
+        [&my_elements_and_cores, my_node, &node_of_elements](
+            const ElementId<Dim>& element_id, const size_t target_proc,
+            const size_t target_node) {
+          node_of_elements.insert(std::pair{element_id, target_node});
+          if (target_node == my_node) {
+            my_elements_and_cores.push_back(std::pair{element_id, target_proc});
+          }
+        },
+        element_weight, blocks, initial_extents, initial_refinement_levels,
+        quadrature,
+
```

Comment:
Remove blank line

Reply:
I personally find the long functions easier to parse when there's a logical split between arguments. In this case everything below is related to parallelization/hardware and the above to the domain.

Reply:
Could you add a comment then instead of a blank line? We do this in some other places and I think it's clearer (a blank line looks like a formatter error to me)